### PR TITLE
Add utility function for kinetic energy computation

### LINKED ISSUE #1272: Isolate the function to compute kinetic energy
- [ ] #1242

### PR DESCRIPTION
**Pull request by akshaysridhar**
### Purpose
Introduces a utility function for kinetic energy computation. 
Closes #1272

### CODE REVIEW

**Comment by akshaysridhar** [comment]
Changes affect MSE tables : Could someone please confirm this is acceptable ?

**Comment by szy21** [comment]
My understanding is that we don't need to merge this now? @Zhengyu-Huang ?

**Comment by akshaysridhar** [comment]
> My understanding is that we don't need to merge this now? @Zhengyu-Huang ?

That's my understanding as well - #1240 is the first in the sequence of PRs we need to merge (the weighted interpolation component changes are separated in that one)

**Comment by tapios** [comment]
Could we isolate the KE computation to a function, so we do not have to repeat this code so many times and can apply the changes necessary once?

**Comment by simonbyrne** [comment]
@akshaysridhar rather than try to merge the commit, use `git cherry-pick` to pull out the single commit from #1234

File: docs/Manifest.toml [hunk 1]
```diff
@@ -134,9 +134,15 @@ version = "0.2.1"
 
 [[deps.CUDA]]
 deps = ["AbstractFFTs", "Adapt", "BFloat16s", "CEnum", "CompilerSupportLibraries_jll", "ExprTools", "GPUArrays", "GPUCompiler", "LLVM", "LazyArtifacts", "Libdl", "LinearAlgebra", "Logging", "Printf", "Random", "Random123", "RandomNumbers", "Reexport", "Requires", "SparseArrays", "SpecialFunctions", "TimerOutputs"]
+<<<<<<< HEAD
+git-tree-sha1 = "6717cb9a3425ebb7b31ca4f832823615d175f64a"
+uuid = "052768ef-5323-5732-b1bb-66c8b64840ba"
+version = "3.13.1"
+=======
```
Comment (simonbyrne) [inline docs/Manifest.toml:141]: Cherry picking the commit will avoid this problem

**Review by simonbyrne — CHANGES_REQUESTED** [review]

**Comment by tapios** [comment]
> Could we isolate the KE computation to a function, so we do not have to repeat this code so many times and can apply the changes necessary once?

We need two things here (or in whatever becomes of this):
1.  Isolate the KE changes to one place (rather than repeating the same changes many times)
2. Document how KE is being computed. (It looks like there is some assumption of orthogonality between horizontal and vertical components, which wouldn't be true in generalized coordinates. I would like to be able to understand what is happening here.)

**Comment by akshaysridhar** [comment]
@simonbyrne Here are the changes related to the `compute_kinetic!` utility; buildkite checks are passing so I'm ready to merge this with your approval.

**Pull request merged**